core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -383,8 +383,9 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
       logInfo("Spark configuration:\n" + _conf.toDebugString)
     }
 
-    // Set Spark driver host and port system properties
-    _conf.setIfMissing("spark.driver.host", Utils.localHostName())
+    // Set Spark driver host and port system properties. This explicitly sets the configuration
+    // instead of relying on the default value of the config constant.
+    _conf.set(DRIVER_HOST_ADDRESS, conf.get(DRIVER_HOST_ADDRESS))
```

Comment:
nit: could you use `_conf` instead? I thought `_conf` and `conf` were different at first glance. But actually they are same.